riscv/csrs.cc

```diff
@@ -618,6 +619,11 @@ reg_t misa_csr_t::dependency(const reg_t val, const char feature, const char dep
 }
 
 bool misa_csr_t::unlogged_write(const reg_t val) noexcept {
+  // the write is ignored if increasing IALIGN would misalign the PC
+  if (!(val & (1L << ('C' - 'A'))) && (state->pc & 2) &&
+      proc->extension_changeable(EXT_ZCA))
+    return false;
```

Comment:
Will this work correctly with `--isa=rv64imafdc`? It seems that `extension_changeable(EXT_ZCA)` will be false in that case, so this would incorrectly allow C to be disabled for that ISA even when the PC is misaligned.

Reply:
`extension_changeable(EXT_ZCA) `will be true in this case. 'C' will imply Zca, so`isa->extension_enabled(EXT_ZCA)` and `isa->extension_enabled('C') `will be true. And `isa->extension_enabled(EXT_ZCB) || isa->extension_enabled(EXT_ZCMP) || isa->extension_enabled(EXT_ZCMT)` will be false in this case.